pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +71,86 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+func FindMissingInputImageStreams(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+OUTER:
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+
+				// note, BuildConfig.Spec.BuildSpec.Strategy.[Docker|Source|Custom]Stragegy.From Input of ImageStream type will get converted to ImageStreamTag on the vX to api conversion
+
+				if !imageStreamNode.IsFound {
+					markers = append(markers, osgraph.Marker{
+						Node:         bcNode,
+						RelatedNodes: []graph.Node{istNode},
+
+						Severity: osgraph.ErrorSeverity,
+						Key:      MissingInputImageStreamErr,
+						Message: fmt.Sprintf("%s builds from %s, but the image stream does not exist.",
+							bcNode.(*buildgraph.BuildConfigNode).ResourceString(), imageStreamNode.ResourceString()),
+						Suggestion: osgraph.Suggestion(fmt.Sprintf("create the image stream %s with the required images", imageStreamNode.ResourceString())),
```

Comment:
actually no, trying to build bc/A will fail is istag/foo doesn't exist, because that means bc/A has an invalid output reference (it's referencing an imagestream that doesn't exist)